File: examples/key/new_managed_certificate/NewManagedCertificate.java
import com.bloock.sdk.client.KeyClient;
import com.bloock.sdk.entity.key.*;

public class NewManagedCertificate {
    public static void main(String[] args) throws Exception {
        KeyClient keyClient = new KeyClient();

        // Load a managed key
        ManagedCertificate key = keyClient.newManagedCertificate(new ManagedCertificateParams(
                KeyType.Rsa2048,
                new SubjectCertificateParams(
                        "BLOOCK",
                        "Organization",
                        "Organization Unit",
                        "Location",
                        "State",
                        "Country"),
                12));
    }
}

File: examples/key/new_managed_certificate/new_managed_certificate.ts
import {
  KeyClient,
  KeyType,
  ManagedCertificateParams,
  SubjectCertificateParams,
} from '@bloock/sdk';

const keyClient = new KeyClient();

// Load a managed key
const _certificate = await keyClient.newManagedCertificate(
  new ManagedCertificateParams(
    KeyType.Rsa2048,
    new SubjectCertificateParams('Bloock'),
    2
  )
);
